### DISCUSSION TITLE
Generate static nested routes with params during build process

### DISCUSSION THREAD

**Discussion by BishalN** [vercel/next.js#51272] [App Router]
## I'm building a blog website which is inspired by https://fasterthanli.me/series 
How can we achieve this in nextjs app router ?
***/series → contains lists of series***
***/series/[slug] → individual series contains list of articles belonging to series***
***/series/[slug]/[blog-slug] → individual blog of a series***
All should be done as SSG 
Any resources and examples would be appreciated

**Comment by icyJoseph** [comment]
This is your friend: https://nextjs.org/docs/app/api-reference/functions/generate-static-params#multiple-dynamic-segments

As for the file structure, you want this:

```console
.
├── README.md
├── next-env.d.ts
├── next.config.js
├── package.json
├── postcss.config.js
├── public
│   ├── next.svg
│   └── vercel.svg
├── src
│   └── app
│       ├── favicon.ico
│       ├── globals.css
│       ├── layout.tsx
│       ├── page.tsx
│       └── series
│           ├── [slug]
│           │   ├── [blog-slug]
│           │   │   └── page.tsx
│           │   └── page.tsx
│           └── page.tsx
├── tailwind.config.js
├── tsconfig.json
└── yarn.lock

7 directories, 16 files
```

Most notably the src/app/series section.

Then you need functions to get all the series, all articles per series, and data from one article. 

You'd use `generateStaticParams` to list all possible `/series/[slug]`, and `series/[slug]/[blog-slug]`, as a list of objects, with keys `{slug: 'something'}` for the former, and `{slug: 'something', 'blog-slug': 'article'}` 

Perhaps consider `blogSlug` or `articleSlug` instead of `-` separated.

Next.js will handle deduping when building the pages.

You can use this one as reference, not as fancy/advanced, but a reference anyway: https://github.com/icyJoseph/icyJoseph/tree/dev/src/app/blog
